CONTRIBUTING.md

```diff
@@ -1,287 +1,232 @@
-
-#### Verify environment is setup correctly
-
-Run a command that is present in the iot extension space
+### [azure-cli](https://github.com/Azure/azure-cli)
 
 ```powershell
-az iot central app -h
+git clone https://github.com/Azure/azure-cli
 ```
 
-If this works, then you should now be able to make changes to the extension and have them reflected immediately in your az cli.
+> IMPORTANT: When cloning the repositories and environments, ensure they are all siblings to each other. This makes things much easier down the line.
+
+```text
+source-directory/
+|-- azure-cli/
+|-- azure-iot-ops-cli-extension/
+|-- .env3/
+```
 
-## Unit and Integration Testing
+To assist in local development, it's suggested to utilize a local clone of the core [azure-cli](https://github.com/Azure/azure-cli) repository.
 
-Tests are organized into folders by resource in `azext_iot\tests\`:
-- Central
-- Digital Twins
-- Device Provisioning Service
-- IoT Hub
-- Product
-- Utility
+You may also utilize `azdev setup` (explained in the next section) to customize your local CLI installation.
 
-### Unit Tests
+## Install and configure [Azure CLI dev tools](https://github.com/Azure/azure-cli-dev-tools)
 
-You may need to install the dev_requirements for this
+With your **virtual environment activated**:
 
-```powershell
-pip install -r path/to/source/dev_requirements
-```
+1.  ```powershell
+    (env) pip install azdev
+    ```
 
-Example unit tests runs:
+2.  ```powershell
+    (env) azdev setup -c ./path_to/azure-cli
+    ```
 
-_Hub:_
-`pytest azext_iot/tests/iothub/core/test_iot_ext_unit.py`
+    `azdev setup` can utilize your local copy of the `azure-cli` repository, or it can download one for you:
+    - `azdev setup -c ./path_to/azure-cli` will use your local copy of the `azure-cli` repo
+    - `azdev setup -c EDGE` will install the current dev branch of the `azure-cli` repo
+    - `azdev setup` with no arguments provides an interactive setup experience
+    - `azdev setup -h` will display other configuration options.
 
-_DPS:_
-`pytest azext_iot/tests/dps/core/test_dps_discovery_unit.py`
+3. Verify your CLI is configured correctly:
 
-Unit tests end in "_unit.py" so execute the following command to run all unit tests,
-`pytest -k "_unit.py"`
+    ```powershell
+    (env) az -v
+    ```
 
-Execute the following command to run the IoT Hub unit tests:
+    You should see `Python location` with a Python executable in your local `env3` path:
 
-`pytest azext_iot/tests/iothub/ -k "_unit.py"`
+    ```powershell
+    Python location 'C:\src\azure-iot-ops-cli-extension\env3\Scripts\python.exe'
+    ```
 
-### Integration Tests
+## Install dev extension
 
-Integration tests are run against Azure resources and depend on environment variables.
+You can isolate your development extensions in a separate folder to aid in debugging.
 
-Example int tests runs:
+Create a new folder for your extensions, and create an environment variable named `AZURE_EXTENSION_DIR` that points to it:
```

Comment:
not sure if we should move this back up - so we can have the example folder structure contain all of the folders + we can add in check the extension directory in `az -v`